vm/systemSmartContracts/eei.go

```diff
@@ -301,6 +301,10 @@ func (host *vmContext) ProcessBuiltInFunction(
 		return errors.New(vmOutput.ReturnMessage)
 	}
 
+	for _, logEntry := range vmOutput.Logs {
```

Comment:
If this code will not be used in mainchain, maybe it is better to move it in the sovereignEei.go file, just to avoid conflicts in the future merges from mainchain last version to sovereign! If it will take to much time now, we can add a TODO here for doing this in the following PRs.

Reply:
This line of code will not be executed on mainchain, it is protected above. at isBuiltInIntraShard. It will yield false(nil vmInput) when running on metachain.

Reply:
Yes, but I still think it would be cleaner to separate it for sovereign?

Reply:
added todo

Reply:
added

Reply:
The code is built in such a way to never call these lines. Like we can add another check. if selfShardID == metachain - than do the transfer only.